Tools/peg_generator/scripts/test_parse_directory.py

```diff
@@ -285,8 +256,7 @@ def main() -> None:
             skip_actions,
             tree,
             short,
-            None,
-            0,
+            1,
```

Comment:
Oh. This seems the problem: by passing `mode=1` here, we will call `parse_string()` (on line 174) which returns an AST. This requires converting the C-level AST (which is allocated using an arena) to a Python-usable AST (built out of Python objects) and that's known to be much slower. Originally this passed `mode=0`, which meant "don't return anything". It seems we no longer have that option, which is fine, but then we should at least pass `mode=2`.

Again, we should refactor a bunch of stuff here so it's more readable.

Reply:
> Again, we should refactor a bunch of stuff here so it's more readable.

I've been working on that. I'll open a new PR for the refactorings, after we've merged this one, if you're okay with that.